dcp-0009/dcp-0009.mediawiki

```diff
@@ -0,0 +1,279 @@
+consensus rules could be retroactively updated to be guarded by the transaction
+version rather than the agenda.  This is desired because it doesn't require
+passing around whether the agenda is active, which itself is based on the
+blockchain global state, and instead could just be based on the transaction
+itself.
+
+===Enforcing Zero Fee for Revocation Transactions===
+
+There must be a rule to limit the revocation transaction fee since the updated
+rules allow tickets to be revoked by anyone.  There is not a strong reason to
+allow a non-zero fee since it will be enforced that blocks MUST contain version
+<code>2</code> ticket revocation transactions.
+
+Further, fees on revocations could potentially create a misalignment of
+incentives in the future when the block rewards are exhausted.  For instance, if
+the amount earned from an additional non-mandatory vote is less than the amount
+that could be earned from a revocation transaction, then miners could be
+incentivised to not include a non-mandatory vote and instead include a
+revocation.
+
+===Ticket Redeemer Output Amounts Calculation===
+
+In order to enforce zero fee for revocation transactions, the ticket redeemer
+output amounts calculation needs to be updated.
+
+As described in the specification section of this proposal, after the original
+contribution amounts are evenly distributed to each output, there may be a
+remainder left over.  This needs to be modified in order to accomodate the
+proposed rule change that requires revocation transaction fees to be zero.
+
+This proposal uses deterministic uniformly pseudorandom selection to select an
+output to receive each atom in the remainder.  A deterministic uniformly
+pseudorandom selection is used to ensure that the output selection cannot be
+gamed.
+
+'''TODO - Add more detail to selection process.'''
+
+The method being used to do the uniform pseudorandom selection was chosen since
+it is well-proven and already in place for the ticket selection lottery.
+
+===Requiring Blocks to Contain Revocation Transactions===
+
+Requiring blocks to contain ticket revocation transactions makes this whole
+revocation process automatic, resulting in a better user experience for
+stakeholders.  The updated rules for revocations described in this proposal
+enable this possibility, since version <code>2</code> ticket revocation
+transactions can be created by anyone, including miners, since they will not
+require a valid signature.
+
+==Deployment==
+
+This agenda will be deployed to mainnet using the standard Decred on-chain
+voting infrastructure as follows:
+
+{|
+!Name!!Setting
+|-
+|Deployment Version||9
+|-
+|Agenda ID||autorevocations
+|-
+|Agenda Description||Enable automatic ticket revocations as defined in DCP0009
+|-
+|Start Time||1631750400 (Sep 16th, 2021 00:00:00 +0000 UTC)
+|-
+|Expire Time||1694822400 (Sep 16th, 2023 00:00:00 +0000 UTC)
+|-
+|Mask||0x06 (Bits 1 and 2)
```

Comment:
This will need to be updated to the correct values since it will be run concurrently with DCP0007 and DCP0008. 

```suggestion
|Mask||0x0060 (Bits 5 and 6)
```

Reply:
Updated.